### PR TITLE
Adding java-springboot-developer.md

### PR DESCRIPTION
**Pull request by jyotsnaravikumar**

### CODE REVIEW

File: docs/Java-SpringBoot-Developer.md [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+    - [Partition Key Function](#partition-key-usage-in-spring)
+4. [AKS Pod Identity Support](#aks-pod-identity-support) - WIP
+5. [Versioning](#versioning)
+6. [Application Insights](#application-insights)
+
+
+## Managed Identity and Key Vault
+
+After creating a Managed Identity for the Helium web app and assigning get and list secret permissions to Key Vault, the following code successfully authenticates using Managed Identity to create the Key Vault Client. Leveraging Managed Identity in this way eliminates the need to store any credential information in app code. 
+
```
Comment (aflinchb) [inline docs/Java-SpringBoot-Developer.md]: We should update the description for managed identity (or somewhere) to better reflect the current solution/state in spring boot.  The description as is says there's no need to store any credential information, but below it's called out that client id and key need to be provided to run locally. 

Maybe (as you suggested earlier), it's best to just remove that first paragraph for now?

File: docs/Java-SpringBoot-Developer.md [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+```properties
+azure.keyvault.enabled=true
+azure.keyvault.uri=https://${KeyVaultName}.vault.azure.net/
+```
+
+### JAVA-SPRINGBOOT-KEYVAULT-SDK-GAP : Local Development issue with spring-boot starter for key-vault with MSI
+This does not works in the local development scenario as the spring-boot keyvault java sdk fails to get Key Vault access through MSI on local development environment​ 
+This is a security hole in development environment which was uncovered here and now is seen with Walmart as well
+
```
Comment (aflinchb) [inline docs/Java-SpringBoot-Developer.md]: Can't use "Walmart"

File: docs/Java-SpringBoot-Developer.md [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+Get an Application Insights instrumentation key by creating an Application Insights resource. Set the application type to Java web application.
+
+Store the application insights instrumentation key in Key Vault to configure Helium to use Application Insights. When configured, the DI creates a singleton instance of TelemetryClient that can be used to track custom events and metrics.
+
+#### Adding Application Insights from Spring framework DI 
+
+Add the dependency "azure-keyvault-secrets-spring-boot-starter" and "azure-client-authentication" to the maven POM file
+
+[POM.xml] (https://github.com/microsoft/helium-java/blob/master/pom.xml) 
+
```
Comment (aflinchb) [inline docs/Java-SpringBoot-Developer.md]: Remove space between [POM.xml] and (https...) so the link renders correctly. 
I.e. : [POM.xml](https://github.com/microsoft/helium-java/blob/master/pom.xml)

File: docs/Java-SpringBoot-Developer.md [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+### Key Vault 
+
+If you need access to Key Vault in your app, you can retrieve the Key Vault Client from Spring framework's DI rather than have to track credentials and create a new connection.
+
+#### Adding Key Vault via Spring Configuration
+
+Add the dependency "azure-keyvault-secrets-spring-boot-starter" and "azure-client-authentication" to the maven POM file
+
+[POM.xml] (https://github.com/microsoft/helium-java/blob/master/pom.xml) 
+
```
Comment (aflinchb) [inline docs/Java-SpringBoot-Developer.md]: Change to: [POM.xml](https://github.com/microsoft/helium-java/blob/master/pom.xml) so link renders properly (remove space)

File: docs/Java-SpringBoot-Developer.md [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+azure.keyvault.enabled=true
+azure.keyvault.uri=https://${KeyVaultName}.vault.azure.net/
+azure.keyvault.client-id=${client_id}
+azure.keyvault.client-key=${client_key}
+
+```
+
+To use managed identities for App Services - please refer to [Using ManagedIdentities setup] (https://docs.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=dotnet)
+
```
Comment (aflinchb) [inline docs/Java-SpringBoot-Developer.md]: Same thing here with the link to remove the space between the ] and (, there are a few throughout.

File: docs/Java-SpringBoot-Developer.md (hunk 1 above)
Comment (jyotsnaravikumar) [reply docs/Java-SpringBoot-Developer.md]: As I had mentioned, this documented is designed in a way similar to CSharp minus the language specifics. So first paragraph is describe the feature implementation from specs, second paragraph is about the gaps in SDK uncovered , feedbacks given and this paragraph is about alternative solutions used due to the gaps. IMO this should stay as that is feature description from specs and we are called out the gaps in the next section. Happy to get feedbacks from others as well.

File: docs/Java-SpringBoot-Developer.md (hunk 2 above)
Comment (jyotsnaravikumar) [reply docs/Java-SpringBoot-Developer.md]: Yes, removing this

File: docs/Java-SpringBoot-Developer.md [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+@Document(collection = Constants.DEFAULT_MOVIE_COLLECTION_NAME)
+@JsonPropertyOrder({"id","movieId", "partitionKey",  "type", "title", "textSearch", "year", "runtime", "rating", "votes", "totalScore", "genres", "roles" })
+public class Movie  extends  MovieBase{
+
+  @JsonIgnore
+  @Id
+  private String id;
+
+  @PartitionKey
+  private String partitionKey;
```
Comment (bartr) [inline docs/Java-SpringBoot-Developer.md]: If spring-boot doesn't use partition key, why do we need it?

File: docs/Java-SpringBoot-Developer.md [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+    //@Query(“Select * from c where c.year= :1 order by order by c.year ”)
+    //List<Movie> findMoviesByYear(int year);
+
+}
+ ```
+
+#### Solution
+
+- Customers are moving to Cosmos Mongo for easier developer experience with features on Native queries, Cross doc queries, paginations, flexibility support
+
```
Comment (bartr) [inline docs/Java-SpringBoot-Developer.md]: We shouldn't be recommending the Mongo API on Cosmos. Does it make sense to remove this section?

File: docs/Java-SpringBoot-Developer.md [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+### Spring Repository Pattern
+
+### JAVA-SPRINGBOOT-COSMOSDB-SDK-GAP: Spring framework does not support different document types in the same collection
+
+This is limitation and needs a spring-boot patch. Spring-Boot entity framework works on entity<-> repo mapping.
+Hence we cannot model movies, genre,actors in the same "movies" collection as recommended in this best practices [document](https://github.com/4-co/imdb)
+spring-data-cosmosdb expects to have 3 separate collections as "movies", "genre", "actor"
+However this is cost implication as "genre" collection having just 30 elements needs a separate collection and is 3 times the RU querying for 3 different collection.
+
+#### Solution
```
Comment (bartr) [inline docs/Java-SpringBoot-Developer.md]: Solution appears as a header multiple times which causes 3 linter errors. Should we ignore the linter error, not use a heading or distinguish the headings?

Comment (aflinchb) [reply docs/Java-SpringBoot-Developer.md]: How about remove the "####" heading and make "Solution" **bold** ?

File: docs/Java-SpringBoot-Developer.md (hunk 1 above)
Comment (aflinchb) [reply docs/Java-SpringBoot-Developer.md]: That works, I also notice now this paragraph only talks about the web app solution.

File: docs/Java-SpringBoot-Developer.md (hunk 6 above)
Comment (jyotsnaravikumar) [reply docs/Java-SpringBoot-Developer.md]: https://github.com/microsoft/spring-data-cosmosdb/blob/master/QueryPartitionedCollection.md 
We are querying by partition key

File: docs/Java-SpringBoot-Developer.md (hunk 7 above)
Comment (jyotsnaravikumar) [reply docs/Java-SpringBoot-Developer.md]: Ok will remove it

File: docs/Java-SpringBoot-Developer.md (hunk 6 above)
Comment (bartr) [reply docs/Java-SpringBoot-Developer.md]: Does that make it faster / less RUs? If so, makes sense to keep it. If not, does it make sense to have that query be "different"?

**Pull request merged**